channel.go

```diff
@@ -183,11 +183,56 @@ func (ch *Channel) Delete() error {
 	return ch.client.makeRequest(http.MethodDelete, p, nil, nil, nil)
 }
 
+type truncateOptions struct {
+	HardDelete  bool       `json:"hard_delete,omitempty"`
+	SkipPush    bool       `json:"skip_push,omitempty"`
+	TruncatedAt *time.Time `json:"truncated_at,omitempty"`
+	Message     *Message   `json:"message,omitempty"`
+}
+
+type TruncateOption func(*truncateOptions)
+
+func WithTruncateHardDelete(hardDelete bool) func(*truncateOptions) {
+	return func(o *truncateOptions) {
+		o.HardDelete = hardDelete
+	}
+}
+
+func WithTruncateSkipPush(skipPush bool) func(*truncateOptions) {
+	return func(o *truncateOptions) {
+		o.SkipPush = skipPush
+	}
+}
+
+func WithTruncateMessage(message *Message) func(*truncateOptions) {
+	return func(o *truncateOptions) {
+		o.Message = message
+	}
+}
+
+func WithTruncateTruncatedAt(truncatedAt *time.Time) func(*truncateOptions) {
```

Comment:
Can we rename option functions to: `TruncateWith*` seems more approriate.
Instead of `WithTruncateTruncatedAt` it will be `TruncateWithTruncatedAt`

Reply:
renamed

Reply:
nice suggestion